lib/internal/crypto/webcrypto.js

```diff
@@ -0,0 +1,751 @@
+}
+
+function exportKeySpki(key) {
+  switch (key.algorithm.name) {
+    case 'RSASSA-PKCS1-V1_5':
+      // Fall through
+    case 'RSA-PSS':
+      // Fall through
+    case 'RSA-OAEP':
+      if (key.type === 'public') {
+        return lazyRequire('internal/crypto/rsa')
+          .rsaExportKey(key, kWebCryptoKeyFormatSPKI);
+      }
+      break;
+    case 'ECDSA':
+      // Fall through
+    case 'ECDH':
+      if (key.type === 'public') {
+        return lazyRequire('internal/crypto/ec')
+          .ecExportKey(key, kWebCryptoKeyFormatSPKI);
+      }
+      break;
+    case 'NODE-DSA':
+      if (key.type === 'public') {
+        return lazyRequire('internal/crypto/dsa')
+          .dsaExportKey(key, kWebCryptoKeyFormatSPKI);
+      }
+      break;
+    case 'NODE-DH':
+      if (key.type === 'public') {
+        return lazyRequire('internal/crypto/diffiehellman')
+          .dhExportKey(key, kWebCryptoKeyFormatSPKI);
+      }
+      break;
+  }
+
+  return PromiseReject(
+    lazyDOMException(
+      `Unable to export a raw ${key.algorithm.name} ${key.type} key`,
+      'InvalidAccessError'));
+}
+
+function exportKeyPkcs8(key) {
+  switch (key.algorithm.name) {
+    case 'RSASSA-PKCS1-V1_5':
+      // Fall through
+    case 'RSA-PSS':
+      // Fall through
+    case 'RSA-OAEP':
+      if (key.type === 'private') {
+        return lazyRequire('internal/crypto/rsa')
+          .rsaExportKey(key, kWebCryptoKeyFormatPKCS8);
+      }
+      break;
+    case 'ECDSA':
+      // Fall through
+    case 'ECDH':
+      if (key.type === 'private') {
+        return lazyRequire('internal/crypto/ec')
+          .ecExportKey(key, kWebCryptoKeyFormatPKCS8);
+      }
+      break;
+    case 'NODE-DSA':
+      if (key.type === 'private') {
+        return lazyRequire('internal/crypto/dsa')
+          .dsaExportKey(key, kWebCryptoKeyFormatPKCS8);
+      }
+      break;
+    case 'NODE-DH':
+      if (key.type === 'private') {
+        return lazyRequire('internal/crypto/diffiehellman')
+          .dhExportKey(key, kWebCryptoKeyFormatPKCS8);
+      }
+      break;
+  }
+
+  return PromiseReject(
+    lazyDOMException(
+      `Unable to export a pkcs8 ${key.algorithm.name} ${key.type} key`,
+      'InvalidAccessError'));
+}
+
+function exportKeyRaw(key) {
+  switch (key.algorithm.name) {
+    case 'ECDSA':
+      // Fall through
+    case 'ECDH':
+      if (key.type === 'public') {
+        return lazyRequire('internal/crypto/ec')
+          .ecExportKey(key, kWebCryptoKeyFormatRaw);
+      }
+      break;
+    case 'AES-CTR':
+      // Fall through
+    case 'AES-CBC':
+      // Fall through
+    case 'AES-GCM':
+      // Fall through
+    case 'AES-KW':
+      // Fall through
+    case 'HMAC':
+      return PromiseResolve(key[kKeyObject].export().buffer);
+  }
+
+  return PromiseReject(
+    lazyDOMException(
+      `Unable to export a raw ${key.algorithm.name} ${key.type} key`,
+      'InvalidAccessError'));
+}
```

Comment:
```suggestion
async function exportKeyRaw(key) {
  switch (key.algorithm.name) {
    case 'ECDSA':
      // Fall through
    case 'ECDH':
      if (key.type === 'public') {
        return lazyRequire('internal/crypto/ec')
          .ecExportKey(key, kWebCryptoKeyFormatRaw);
      }
      break;
    case 'AES-CTR':
      // Fall through
    case 'AES-CBC':
      // Fall through
    case 'AES-GCM':
      // Fall through
    case 'AES-KW':
      // Fall through
    case 'HMAC':
      return key[kKeyObject].export().buffer;
  }

  throw lazyDOMException(
    `Unable to export a raw ${key.algorithm.name} ${key.type} key`,
    'InvalidAccessError');
}
```
this way the function will always be async, and you won't be forced to remember to return a promise. (i'm also assuming you don't want this function to be z̲̗̼͙̥͚͛͑̏a̦̟̳͋̄̅ͬ̌͒͟ļ̟̉͌ͪ͌̃̚g͔͇̯̜ͬ̒́o̢̹ͧͥͪͬ for some concrete reason)